Wrappers/Python/test/test_AdjointOperator.py

```diff
@@ -0,0 +1,107 @@
+
+from utils import initialise_tests
+from cil.framework import ImageGeometry
+from cil.optimisation.operators import GradientOperator, AdjointOperator, MatrixOperator
+import numpy as np 
+
+
+from testclass import CCPiTestClass
+
+
+initialise_tests()
+
+class TestAdjointOperator(CCPiTestClass):
+    
+    def setUp(self):
+        self.ig_real = ImageGeometry(3,4)
+    
+    def tearDown(self):
+        pass   
+    
+    def test_direct_adjoint_gradient_operator(self):
+
+        G = GradientOperator(self.ig_real)
+        x = G.domain_geometry().allocate("random", seed=10)
+        y = G.range_geometry().allocate("random", seed=10)
+        G_adjoint = AdjointOperator(G)
+
+        res1 = G.adjoint(y)
+        res2 = G_adjoint.direct(y)
+        np.testing.assert_allclose(res1.array, res2.array, atol=1e-3)
+
+        res3 = G.direct(x)
+        res4 = G_adjoint.adjoint(x)
+        np.testing.assert_allclose(res3[0].array, res4[0].array, atol=1e-3)
```

Comment:
I wonder if this tolerance of 1e-3 is too loose? These should be numerically identical? Some should-be fails might slip through?

Reply:
Changed to 1e-6